### PR TITLE
fix(api-client): use base server URL in the client

### LINKED ISSUE #3606: baseServerURL doesn’t work
I’m passing `baseServerURL` in the reference configuration, but it doesn’t change anything in the base URL in the reference or in the client.

### PR DESCRIPTION
**Pull request by amritk**
closes #3606

This is what we wanted right @hanspagel ?

### CODE REVIEW

File: packages/api-reference/src/features/BaseUrl/ServerForm.vue [hunk 1]
```diff
@@ -43,7 +43,7 @@ function updateVariable(name: string, value: string) {
 
 <template>
   <div v-if="servers?.length">
-    <label class="server-form-title">Base URL</label>
+    <label class="server-form-title">Server URL</label>
```
Comment (amritk) [inline packages/api-reference/src/features/BaseUrl/ServerForm.vue:46]: I also renamed Base URL -> Server URL in the UI so there's less confusion.

**Comment by hanspagel** [comment]
I think it doesn’t work in the references aswell, need to test later. ✌️

**Comment by hanspagel** [comment]
Okay, it works in the reference, but not in the client.

**OpenAPI document**
```json
        {
            "openapi": "3.1.0",
            "info": {
                "title": "Hello World",
                "version": "1.0.0",
            },
            "servers": [{
                    "url": "/"
                },
            ],
            "paths": {
                "/foobar": {
                    "get": {
                        "operationId": "getAllData",
                        "responses": {
                            "200": {
                                "description": "OK",
                                "content": {
                                    "application/json": {
                                        "schema": {
                                          "type": "object",
                                          "properties": {
                                              "message": {
                                                  "type": "string",
                                                  "example": "Hello World"
                                              }
                                          }
                                        }
                                    }
                                }
                            }
                        }
                    },
                },
            },
        }
```

**Configuration**

```js
{
    baseServerURL: 'https://something-custom.scalar.com',
}
```

**Result**

✅ 

Screenshot 2024-11-06 at 13 59 13

❌

Screenshot 2024-11-06 at 13 59 16

**Comment by amritk** [comment]
@hanspagel ur right, I swear it was working before 😓

**Comment by amritk** [comment]
alright @hanspagel fixed! maybe I dropped some changes before 🤷🏾

**Review by hanspagel — CHANGES_REQUESTED** [review]
This is great and we’re so close! But for some reason it has only one slash. 🙈

Screenshot 2024-11-07 at 11 56 30

Screenshot 2024-11-07 at 11 57 22

**Comment by amritk** [comment]
Hahaha duh I'm stupid

**Comment by amritk** [comment]
@hanspagel alright this is the one, tests have been added!

**Review by marclave — APPROVED** [review]

**Pull request merged**